File: FactoryEmulatorAndServer/model/pump.py
import copy


class Pump:
    def __init__(self, tag, allowVoltage, allowTurnovers, tankSource, tankDirection):
        self.voltage = 0
        self.turnovers = 0
        self.allowVoltage = allowVoltage
        self.allowTurnovers = allowTurnovers
        self.tag = tag

        self.tankSource = tankSource
        self.tankDirection = tankDirection

    def pump(self, timeDelta):
        intake = self.turnovers * timeDelta

        intake = min(self.tankSource.fill, intake)

        self.tankSource.fill -= intake
        self.tankDirection.fill += intake

    def turnOn(self, voltage, turnovers):
        self.turnOn = True
        self.voltage = voltage
        self.turnovers = turnovers

    def turnOff(self):
        self.turnOn = False
        self.voltage = 0
        self.turnovers = 0

    def toDict(self):
        d = copy.copy(self.__dict__)
        d['tankSource'] = self.tankSource.tag
        d['tankDirection'] = self.tankDirection.tag
        return d

File: FactoryInMobile/app/src/main/java/com/factory/andre/factoryinmobile/model/Pump.kt
package com.factory.andre.factoryinmobile.model

import com.factory.andre.factoryinmobile.model.base.IFactoryItemsModel
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Pump(
        override var factory: String? = null,

        @SerializedName("tag")
        @Expose
        override var tag: String? = null,


        @SerializedName("voltage")
        @Expose
        var voltage: Float? = null,

        @SerializedName("allowVoltage")
        @Expose
        var allowVoltage: Float? = null,


        @SerializedName("turnovers")
        @Expose
        var turnovers: Float? = null,

        @SerializedName("allowTurnovers")
        @Expose
        var allowTurnovers: Float? = null,

        @SerializedName("turnOn")
        @Expose
        var turnOn: Boolean? = null,

//        TODO add Pump.status to restApi.to realm
        @SerializedName("status")
        @Expose
        var status: String? = null,

        override var timestamp: Long? = null

) : IFactoryItemsModel {
}